src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -599,6 +669,75 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             }
         }
 
+        // This fills in the arguments and parameters arrays in evaluation order.
+        private static void BuildExplicitArgumentsInEvaluationOrder(
+            SyntaxNode syntax,
+            Symbol methodOrIndexer,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<BoundExpression> arguments,
+            Binder binder,
+            /* out */ ArrayBuilder<IArgument> argumentsBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> missingParametersBuilder)
+        {
+            ImmutableArray<ParameterSymbol> parameters = methodOrIndexer.GetParameters();
+            ArrayBuilder<int> processedParameters = ArrayBuilder<int>.GetInstance(parameters.Length);
+
+            for (int a = 0; a < arguments.Length; ++a)
+            {
+                BoundExpression argument = arguments[a];
+
+                int p = (!argsToParamsOpt.IsDefault) ? argsToParamsOpt[a] : a;
+                var parameter = parameters[p];
+
+                Debug.Assert(!processedParameters.Contains(p));
+
+                processedParameters.Add(p); 
+
+                bool isParamArray = false;
+                if (expanded && p == parameters.Length - 1)
```

Comment:
>if (expanded && p == parameters.Length - 1) [](start = 16, length = 43)

The following comment from the previous review pass is not addressed ([https://github.com/dotnet/roslyn/pull/19237#pullrequestreview-37211265](https://github.com/dotnet/roslyn/pull/19237#pullrequestreview-37211265))
>It looks like the logic in this statement (condition + body) is a clone of logic from BuildStoresToTemps. We should find a way to extract the logic from BuildStoresToTemps into a common helper and share it with this method. This will ensure that the logic in both methods doesn't diverge. For example, the extracted method could be called IsBeginningOfParamArray and it could return the number of items in the array or 0 for a regular argument. Based on the number returned from the helper, this method could build the right IArgument instance.